test/extensions/filters/http/lua/lua_integration_test.cc

```diff
@@ -224,17 +269,37 @@ class LuaIntegrationTest : public testing::TestWithParam<Network::Address::IpVer
     }
   }
 
+  const char* expectedDownStreamProtocol() const {
+    switch (downstream_protocol_) {
+    case Http::CodecType::HTTP1:
+      return "HTTP/1.1";
+    case Http::CodecType::HTTP2:
+      return "HTTP/2";
+    case Http::CodecType::HTTP3:
+      return "HTTP/3";
+    default:
+      return "";
+    }
+  }
+
   FakeHttpConnectionPtr fake_lua_connection_;
   FakeStreamPtr lua_request_;
 };
 
-INSTANTIATE_TEST_SUITE_P(IpVersions, LuaIntegrationTest,
-                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
-                         TestUtility::ipTestParamsToString);
+INSTANTIATE_TEST_SUITE_P(Protocols, LuaIntegrationTest,
+                         testing::ValuesIn(LuaIntegrationTest::getDefaultTestParams()),
+                         UpstreamDownstreamIntegrationTest::testParamsToString);
 
 // Regression test for pulling route info during early local replies using the Lua filter
 // metadata() API. Covers both the upgrade required and no authority cases.
 TEST_P(LuaIntegrationTest, CallMetadataDuringLocalReply) {
+  if (!testing_downstream_filter_) {
+    GTEST_SKIP() << "This is a local reply test that does not go upstream";
```

Comment:
For this and many of the other tests that don't make sense for an upstream filter, what is the behavior if a lua script were written that tries to do these things?

Reply:
For local reply (like this test) or direct response tests, the request never makes it upstream and hence upstream filter won't be invoked.

Another test i saw tests clearing route cache, that also can't happen on an upstream filter (it's a no-op in lua, per previous @wbpcode PR).